web/app/js/components/NetworkGraph.jsx

```diff
@@ -0,0 +1,198 @@
+import _ from 'lodash';
+import { metricsPropType } from './util/MetricUtils.js';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withContext } from './util/AppContext.jsx';
+import withREST from './util/withREST.jsx';
+import * as d3 from 'd3';
+import 'whatwg-fetch';
+
+
+const defaultSvgWidth = 524;
+const defaultSvgHeight = 325;
+const margin = { top: 0, right: 0, bottom: 10, left: 0 };
+
+const simulation = d3.forceSimulation()
+  .force("link",
+    d3.forceLink()
+      .id(d => d.id)
+      .distance(140))
+  .force("charge", d3.forceManyBody().strength(-20))
+  .force("center", d3.forceCenter(defaultSvgWidth / 2, defaultSvgHeight / 2));
+
+export class NetworkGraphBase extends React.Component {
+  static defaultProps = {
+    deployments: []
+  }
+
+  static propTypes = {
+    data: PropTypes.arrayOf(metricsPropType.isRequired).isRequired,
+    deployments: PropTypes.arrayOf(PropTypes.object),
+  }
+
+  constructor(props) {
+    super(props);
+  }
+
+  componentDidMount() {
+    this.svg = d3.select(".network-graph-container")
+      .append("svg")
+      .attr("class", "network-graph")
+      .attr("width", defaultSvgWidth)
+      .attr("height", defaultSvgHeight)
+      .append("g")
+      .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+  }
+
+  componentDidUpdate() {
+    simulation.alpha(1).restart();
+    this.drawGraph();
+  }
+
+  getGraphData() {
+    const { data } = this.props;
+    let links = [];
+    let nodeList = [];
+
+    _.map(data, (r, i) => {
```

Comment:
nit: I'd maybe call the variable `r` `rsp`/`statTable`/`datum` (or something else!) instead, since we have another `r` that means something else in line 70